tests/failpoints/cases/test_titan.rs

```diff
@@ -0,0 +1,161 @@
+            .unwrap()
+    );
+    // lv6: file0 [k1: ref_to_blob_file, k3: v]
+    // blob db: file0 [k1: v]
+
+    // Restart node 3 with titan in fallback mode
+    cluster.cfg.rocksdb.defaultcf.titan.blob_run_mode = BlobRunMode::Fallback;
+    cluster.cfg.rocksdb.defaultcf.titan.discardable_ratio = 1.0;
+    cluster.stop_node(3);
+    cluster.restart_engine(3);
+    cluster.run_node(3).unwrap();
+    // Check blob file is deleted
+    let mut blob_file_reclaimed = false;
+    let db = cluster.engines[&3].kv.as_inner();
+    let defaultcf = db.cf_handle(CF_DEFAULT).unwrap();
+    for _i in 0..10 {
+        if db
+            .get_property_int_cf(defaultcf, "rocksdb.titandb.num-live-blob-file")
+            .unwrap()
+            == 0
+        {
+            blob_file_reclaimed = true;
+            break;
+        }
+        sleep_ms(100);
+    }
+    assert!(blob_file_reclaimed);
+    cluster.engines[&3].kv.flush_cf(CF_DEFAULT, true).unwrap();
+    assert_eq!(
+        1,
+        db.get_property_int_cf(defaultcf, "rocksdb.num-files-at-level0")
+            .unwrap()
+    );
+    cluster.engines[&3]
+        .kv
+        .compact_files_in_range_cf(CF_DEFAULT, None, None, Some(5))
+        .unwrap();
+    assert_eq!(
+        1,
+        db.get_property_int_cf(defaultcf, "rocksdb.num-files-at-level6")
+            .unwrap()
+    );
+    assert_eq!(
+        1,
+        db.get_property_int_cf(defaultcf, "rocksdb.num-files-at-level5")
+            .unwrap()
+    );
+    // lv5: file1 [k1: deleted]
+    // lv6: file0 [k1: ref_to_blob_file, k3: v]
+    // blob db: file0 [k1: v]
+
+    cluster.cfg.rocksdb.titan.enabled = Some(false);
+    cluster.stop_node(3);
+    cluster.restart_engine(3);
+    cluster.run_node(3).unwrap();
+    // Deletion record written by blob file GC is deleted by DeleteFiles, while the
+    // original record that is suppose to be deleted by DeleteByWriter is not
+    // deleted.
+    fail::cfg("after_delete_files_in_range", "return()").unwrap();
+    let peer = region1
+        .get_peers()
+        .iter()
+        .find(|p| p.get_store_id() == 3)
+        .unwrap()
+        .clone();
+    assert!(!is_error_response(&block_on(
+        cluster
+            .async_remove_peer(region1.get_id(), peer.clone())
+            .unwrap(),
+    )));
+    cluster.must_remove_region(3, region1.get_id());
+    let db = cluster.engines[&3].kv.as_inner();
+    let defaultcf = db.cf_handle(CF_DEFAULT).unwrap();
+    assert_eq!(
+        0,
+        db.get_property_int_cf(defaultcf, "rocksdb.num-files-at-level5")
+            .unwrap()
+    );
+    assert_eq!(
+        1,
+        db.get_property_int_cf(defaultcf, "rocksdb.num-files-at-level6")
+            .unwrap()
+    );
+    // lv5: file1 got deleted, since it is fully covered by region1
+    // lv6: file0 [k1: ref_to_blob_file, k3: v]
+    // blob db: file0 [k1: v]
+    assert!(!is_error_response(&block_on(
+        cluster
+            .async_add_peer(region1.get_id(), peer.clone())
```

Comment:
Nit: It's better to initiate a transfer leader to the new peer after `async_add_peer`. Because it does not wait for the new peer to be created.